### PR TITLE
Change Client Initial size to 1200 flat

### LINKED ISSUE #684: Simplify Client Initial size
The consensus at the hackathon was that the rules around size of the initial packet were overly onerous.  We agreed at the table that a flat 1200 floor was easiest and equally effective.

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #684.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -2206,19 +2206,16 @@ An endpoint MUST NOT reduce their MTU below this number, even if it receives
...
 
 Clients MUST ensure that the first packet in a connection, and any
-retransmissions of those octets, has a QUIC packet size of least 1232 octets for
-an IPv6 packet and 1252 octets for an IPv4 packet.  In the absence of extensions
-to the IP header, padding to exactly these values will result in an IP packet
-that is 1280 octets.
+retransmissions of those octets, has a QUIC packet size of least 1200 octets.
 
-The initial client packet SHOULD be padded to exactly these values unless the
+The initial client packet SHOULD be padded to exactly 1200 octets unless the
```
Comment (ekr) [inline draft-ietf-quic-transport.md]: Can you make clear that this is just the QUIC packet, not the headers.

Comment (martinthomson) [inline draft-ietf-quic-transport.md]: Done.

File: draft-ietf-quic-tls.md [hunk 2]
```diff
@@ -541,7 +541,7 @@ older than 1.3 is negotiated.
 
 QUIC requires that the initial handshake packet from a client fit within the
 payload of a single packet.  The size limits on QUIC packets mean that a record
-containing a ClientHello needs to fit within 1197 octets.
+containing a ClientHello needs to fit within 1165 octets.
```
Comment (ianswett) [inline draft-ietf-quic-tls.md]: To be clear, this 1165 number may change if we increase or decrease the amount of framing overhead QUIC adds.  Are we ok about that?

Comment (martinthomson) [inline draft-ietf-quic-tls.md]: I'm OK with that.  As long as we stop changing the overhead when we ship the spec.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -2206,19 +2206,17 @@ An endpoint MUST NOT reduce their MTU below this number, even if it receives
...
-reduce the amplitude of amplification attacks caused by server responses toward
-an unverified client address.
+The initial client packet SHOULD be padded to exactly 1200 octets unless the
+client has a reasonable assurance that the PMTU is larger.  This size includes
+the QUIC header and integrity check, but not the UDP or IP header. Sending a
+packet of this size ensures that the network path supports an MTU of this size
+and helps reduce the amplitude of amplification attacks caused by server
+responses toward an unverified client address.
 
 Servers MUST ignore an initial plaintext packet from a client if its total size
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: It'd be nice to come up with a consistent way of referring to this "total size" which does not include IP+UDP.

File: draft-ietf-quic-tls.md (hunk 2 above)
Comment (ianswett) [inline draft-ietf-quic-tls.md]: One q: How are you arriving at 1165 if the 1200 doesn't include IP + UDP?  1(type byte) + 8(conn id) + 4(packet number) + stream frame type byte(1) + stream id(1 byte) + ?

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: "QUIC packet size" or "packet size" should do.

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -1140,7 +1140,7 @@ max_packet_size (0x0005):
   the endpoint is willing to receive, encoded as an unsigned 16-bit integer.
   This indicates that packets larger than this limit will be dropped.  The
   default for this parameter is the maximum permitted UDP payload of 65527.
-  Values below 1252 are invalid.  This limit only applies to protected packets
+  Values below 1200 are invalid.  This limit only applies to protected packets
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: Can we somehow clarify this size is also without UDP + IP headers?

Comment (ianswett) [inline draft-ietf-quic-transport.md]: Payload size?

**Comment by martinthomson** [comment]
@ianswett,

> One q: How are you arriving at 1165 if the 1200 doesn't include IP + UDP? 1(type byte) + 8(conn id) + 4(packet number) + stream frame type byte(1) + stream id(1 byte) + ?

1200- all of that(15) - version(4) - fnv-1a(8), which I realize now is 29 and not 35 because I calculated the original number with FNV-1a 128.

**Pull request merged**